lib/datasource/index.ts

```diff
@@ -55,25 +59,139 @@ const datasources: Record<string, Datasource> = {
   terraform,
 };
 
-const cacheNamespace = 'datasource-releases';
+export function baseUrlLegacyMassager(sourceUrl) {
+  let url: string = sourceUrl.trim();
 
-async function fetchReleases(
+  if (url.startsWith('git@')) {
+    url = 'ssh://' + url; // workaround for https://github.com/npm/hosted-git-info/issues/56
+  }
+  const parsedUrl = URL.parse(url);
+  const extraBaseUrls = [];
+  const getHostsFromRulesGithub = hostRules.hosts({ hostType: 'github' }) || [];
+  // istanbul ignore if
+  if (
+    getHostsFromRulesGithub.includes(parsedUrl.hostname) ||
+    (parsedUrl.hostname &&
+      (parsedUrl.hostname === 'github.com' ||
+        parsedUrl.hostname === 'www.github.com'))
```

Comment:
Is `www.github.com` valid or necessary? I think they only use `github.com`?

Reply:
The www handling was migrated from the way the npm data-source handles the sourceUrl.

Reply:
The www handling and wrangling were part of the url parsing in the npm datasource which I migrated here.